x-pack/plugin/logsdb/qa/with-basic/src/javaRestTest/java/org/elasticsearch/xpack/logsdb/LogsdbRestIT.java

```diff
@@ -72,4 +73,40 @@ public void testFeatureUsageWithLogsdbIndex() throws IOException {
         }
     }
 
+    @SuppressWarnings("unchecked")
+    public void testLogsdbIndexGetsStoredSource() throws IOException {
+        final String index = "test-index";
+        createIndex(index, Settings.builder().put("index.mode", "logsdb").build());
+        Map<String, Object> settings = (Map<String, Object>) ((Map<String, Object>) getIndexSettings(index).get(index)).get("settings");
+        assertEquals("logsdb", settings.get("index.mode"));
+        assertEquals(SourceFieldMapper.Mode.STORED.toString(), settings.get("index.mapping.source.mode"));
+    }
+
+    @SuppressWarnings("unchecked")
+    public void testLogsdbOverrideSyntheticSourceModeInMapping() throws IOException {
+        final String index = "test-index";
+        String mapping = """
+            {
+                "_source": {
+                    "mode": "synthetic"
+                }
+            }
+            """;
+        createIndex(index, Settings.builder().put("index.mode", "logsdb").build(), mapping);
+        Map<String, Object> settings = (Map<String, Object>) ((Map<String, Object>) getIndexSettings(index).get(index)).get("settings");
+        assertEquals("logsdb", settings.get("index.mode"));
+        assertEquals(SourceFieldMapper.Mode.STORED.toString(), settings.get("index.mapping.source.mode"));
+    }
+
+    @SuppressWarnings("unchecked")
+    public void testLogsdbNoOverrideSyntheticSourceSetting() throws IOException {
+        final String index = "test-index";
+        createIndex(
+            index,
+            Settings.builder().put("index.mode", "logsdb").put("index.mapping.source.mode", SourceFieldMapper.Mode.SYNTHETIC).build()
+        );
+        Map<String, Object> settings = (Map<String, Object>) ((Map<String, Object>) getIndexSettings(index).get(index)).get("settings");
+        assertEquals("logsdb", settings.get("index.mode"));
+        assertEquals(SourceFieldMapper.Mode.SYNTHETIC.toString(), settings.get("index.mapping.source.mode"));
```

Comment:
This is an interesting one. We'll need a way to avoid this, eventually.

Reply:
I don't think I follow why synthetic is returned as mode here. Do you know why?
Does the `SyntheticSourceIndexSettingsProvider` not kick in?

Reply:
Seems like a settings merging issue.. I'll take a closer look.

Reply:
Indeed, it's this line: https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateIndexService.java#L1047

So, we first call `additionalIndexSettings` and then merge the request template. The latter wins in case of a conflic, which means that a user may bypass the value that gets injected by licensing.

I suggest we get this in as is and think about how to solve this separately.

Reply:
Sounds like either we disallow setting the index setting or we need a new API in index setting provider?

Reply:
So I missed this detail 🤦 

> or we need a new API in index setting provider?

Yes, I think we need something extra in the `IndexSettingsProvider` interface to indicate that that returned settings are not default but should overwrite whatever has been provided in template or create index request.

Reply:
But let's get this in first and adjust `IndexSettingsProvider` infrastructure later.

Reply:
Yeah that or use inheritance, or prohibit setting this in the request. I'll explore these separately.

Reply:
> or prohibit setting this in the request.

The `index.mapping.source.mode` index setting will be the only way how to configure synthetic source. If we don't want to users to specify this setting then we need to make this a private/hidden setting and then introduce another setting that enables synthetic source. The private/hidden setting will then take preference. This could work as well, I think. Let's which is the easiest way to resolve this issue.